Sources/StreamChat/WebSocketClient/Events/MessageEvents.swift

```diff
@@ -4,78 +4,212 @@
 
 import Foundation
 
-public struct MessageNewEvent: MessageSpecificEvent {
-    public let userId: UserId
-    public let cid: ChannelId
-    public let messageId: MessageId
+/// Triggered when a new message is sent to channel.
+public struct MessageNewEvent: ChannelSpecificEvent, HasUnreadCount {
+    /// The user who sent a message.
+    public let user: ChatUser
+    
+    /// The message that was sent.
+    public let message: ChatMessage
+    
+    /// The channel identifier the message was sent to.
+    public var cid: ChannelId { channel.cid }
+    
+    /// The channel a message was sent to.
+    public let channel: ChatChannel
```

Comment:
That is disputable change 🙂 On the one hand, `channel` does not come with `message.new` event payload. But to start getting `message.new` events, a user has to start watching a channel and the REST API response returns the channel which we save to the database. So at the moment the event arrives, the channel must already exist locally

*On Android, only `cid` is exposed

Reply:
The same applies for `message.updated/message.deleted/message.read`.This change is a separate commit so it's easy to roll it back.

Reply:
What if it is like a really old message that was updated? 🤔 We might not have the channel in the DB in that case right?

Reply:
From what I know, we do not get `message.new/message.updated/message.deleted` unless user starts watching a channel by the API call which returns the channel payload in the response (which we save to db).

Reply:
@nuno-vieira the issue I can imagine is related to race conditions. When we start watching a channel and `message.new/message.updated/message.deleted` come before we received the REST API response with a channel, there's no channel in db -> `eventDTO -> event` conversion fails -> event is not published.

Is it OK not to publish the event in this case? From customer perspective, unless `watch` completion is called the watching hasn't yet started so they do not expect to be notified about the channel events so it seems fine.

cc @tbarbugli @b-onc @adamrushy